client/components/FormattingBar/controlComponents/ControlsFootnoteCitation/ControlsFootnoteCitation.tsx

```diff
@@ -63,20 +64,26 @@ const ControlsFootnoteCitation = (props: Props) => {
 	const { editorChangeObject, onClose, pendingAttrs } = props;
 	const { selectedNode } = editorChangeObject;
 	const { noteManager } = usePubContext();
-	// @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
-	const isFootnote = selectedNode.type.name === 'footnote';
+	const isFootnote = selectedNode?.type?.name === 'footnote';
 	const { commitChanges, hasPendingChanges, updateAttrs: rawUpdateAttrs, attrs } = pendingAttrs;
 	const { structuredValue, unstructuredValue, customLabel } = unwrapPendingAttrs(
 		attrs,
 		isFootnote,
 	);
+
 	const updateAttrs = wrapUpdateAttrs(rawUpdateAttrs, isFootnote);
 	const [renderedStructuredValue, setRenderedStructuredValue] = useState(
 		noteManager.getRenderedValueSync(structuredValue),
 	);
-	const [debouncedValue] = useDebounce(structuredValue, 250);
 	const html = renderedStructuredValue?.html;
 	const showPreview = html || unstructuredValue;
+	const [debouncedValue] = useDebounce(structuredValue, 250);
+
+	const onSelectSuggestedCitation = (selected) => {
+		if (selected.bibtex) {
```

Comment:
We should create a type alias for this object that we can also use on the API side after fetching the results from Zotero. Is it something like `ZoteroCSLJSON & { bibtex: string }`?